### ISSUE TITLE
Prompfoo giving false-positives?

### ISSUE THREAD

**Issue by matthieudelaro** [promptfoo/promptfoo#345]
Hello,

Here is a simple example with two obvious errors.
Yet Promptfoo claims `Failures: 0`

Demonstration: `npx promptfoo@latest eval`:
![image](https://github.com/promptfoo/promptfoo/assets/2452725/cf4cffc9-3a57-441d-9a01-625b921daca7)

Code:
prompt.txt
```
{{question}}
```

promptfooconfig.yaml
```
prompts: prompt.txt # prompt.txt is just a file that contains {{question}}, since we're passing the question directly through to the provider.
providers:
  - exec:python -c "print('[FAIL] 9691')"

# These test properties are applied to every test
defaultTest:
  assert:
    - type: not-contains
      value: returned status code
    - type: not-contains
      value: "Error: "
    - type: not-contains
      value: "[FAIL]"
    - type: not-contains
      value: "Error: "

tests:
  - vars:
      question: 19382/2?
    assert:
      - type: contains
        value: 9691
  - vars:
      question: Hello. Please tell me 'Hello'.
    assert:
      - type: contains
        value: Hello
```

Did I miss or misunderstand something?
My end goal is to test a python script in CI/CD, and rely on `Failures: 0` to validate tests. But since it's always `Failures: 0`...

Thanks for your answers,

**Comment by typpo** [comment]
Hi @matthieudelaro,

Thanks for the catch.  The script invocation is failing due to a syntax error (which incidentally is related to the escaping issue solved in #343).  The failure counter isn't being incremented when the shell command itself throws an error.

This is fixed in 61d1b068f26c63f3234dc49c9d5f5104b9cf1cda and will be resolved in the next published release.

**Comment by typpo** [comment]
Resolved in version 0.33.0

**Comment by matthieudelaro** [comment]
Perfect, thanks for solving that so quickly 🙏
I ran the same test : works like a charm.
